roles/acme/tasks/preconditions.yml

```diff
@@ -6,6 +6,13 @@
       - acme_challenge_provider != ""
     fail_msg: You need to set acme_challenge_provider with a provider. See documentation for a list of possible providers.
 
+- name: Check if a acme_domain is set
+  ansible.builtin.assert:
+    that:
+      - acme_domain is defined
+      - acme_domain != []
```

Comment:
**suggestion:** `acme_domain` is used as a hash, here the check is looking for an empty array. This is clone enough, that there is no missmatch in Ansible, but I think it would be cleaner to handle it as a `{}`.

Also applies to the defaults.